src/averageAges.js

```diff
@@ -55,7 +82,32 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const arrOfDiff = [];
+  const men = people.filter(person => person.sex === 'm');
+
+  if (onlyWithSon) {
```

Comment:
Refactor the `if (onlyWithSon)` block and the `else` block to avoid code duplication.

Reply:
consider using ternary operator instead of `if/else`